### ISSUE TITLE
Passing a nested struct to a function generate an error in spirv-val : type does not match parameter type

### ISSUE THREAD

**Issue by nicovize** [KhronosGroup/glslang#1719]
Hello,

A similar issue has been solved in the past for the case of an indexed struct in a constant buffer, thanks again for that :)

However, there is a similar issue that remains when accessing a nested struct (not necessarily inside an array, this time).

I get this message from spirv-val:

error: line 42: OpFunctionCall Argument <id> '21[%21]'s type does not match Function <id> '7[%F]'s parameter type.
  %22 = OpFunctionCall %void %func_struct_F_f11_ %21

The following compute shader shows the problem :

```
#version 450

struct F
{
	float dummy;
};

struct G
{
	F f;
};

void func(const F f)
{
}

layout(std140, set=0, binding=0) uniform CB
{
	G g;
};

layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
void main() 
{
	func(g.f);
}


```

and the command line I used for validation was :

```
glslangValidator -o nested_struct_passed_to_function.spv -V --target-env vulkan1.1 nested_struct_passed_to_function.comp
spirv-val nested_struct_passed_to_function.spv
```

glslangvalidator version infos  (version from 1.1.101.0 Vulkan SDK) :

```
Glslang Version: 7.11.3113
ESSL Version: OpenGL ES GLSL 3.20 glslang Khronos. 11.3113
GLSL Version: 4.60 glslang Khronos. 11.3113
SPIR-V Version 0x00010300, Revision 6
GLSL.std.450 Version 100, Revision 1
Khronos Tool ID 8
SPIR-V Generator Version 7
GL_KHR_vulkan_glsl version 100
ARB_GL_gl_spirv version 100

```

best regards

**Comment by johnkslang** [comment]
Interesting, it works without the `const` on the function parameter, so this is a case where the right type changes were done to pass-by-copy, but the `const` turns into pass-by-reference and bypasses the copy that corrects the type.

I will look into it. In the meantime, if you don't need the `const`...

**Comment by nicovize** [comment]
thanks a lot for the quick reply! Yes, removing the "const" through a macro or locally in our codebase will do it for now :)
best regards

**Comment by arcady-lunarg** [comment]
This has since been fixed, with the function call getting the correct type passed to it. The commit that appears to have fixed it appears to be 24dcbd1b1f68ff268bfa22dc4b72bec73d52abe7.
